ArgParse and falgs

nasrsaab opened the pull request:
to run the project please write:
python git_parser.py -o <File name> -r <Repo>

nasr-saab committed:
Use argparse and options to work with flags
--output_file = Path on the computer to save the created excel
--repository  = Path to the repository I want the script to parse
 run command as follow: 'python git_parser.py --output_file <File name>
 --repository <Repo>'

nasr-saab committed:
Delete the date
to write all the commits in the excel file
without paying attention to a spcific date

royno-nvidia commented on git_parser.py:
```diff
@@ -3,25 +3,32 @@
-#import pandas as pd
+import argparse
+
+parser= argparse.ArgumentParser(description='git_parser commit from GitHub')
```
Use inside function, call it from 'main'

royno-nvidia commented on git_parser.py:
```diff
@@ -43,4 +50,5 @@ def show_commit(url):
 
-
+if __name__=='__main__':
+    show_commit(args.output_file, args.repository)
```
Pass args as is - what if we would like to add more flags?
Always thing further uses... we need to write scalable code.

royno-nvidia commented on git_parser.py:
```diff
@@ -39,7 +36,7 @@ def show_commit(output_file,repository):
 
     Row_number += 1
 
-    for commit in Repository(path_to_repo=repository, since=firstDay, to=toDay).traverse_commits():
```
Do not remove date use.
Instead lets add new flag call '--show_lats_days' it will get the number of days I want to parse commits from today and backwards.

Example: --show_lats_days=15  will show me only commits from the last 15 days.
if not provided - show all commits In the repository.

royno-nvidia commented:
Please see the review notes.
Thanks

nasr-saab committed:
No duplicate hash commit
today I reback the date and git commit in x last days
and remove a duplicated hash and set the differnt files in the same
culmon in the excel file

nasr-saab committed:
New setup file for the git_parser.py
dont use the module.py file because I set in the git_parser file 'main'
Functin

nasr-saab committed:
Expandable code

nasr-saab committed:
New packages in setup file
beacause I used date a getparse packages

nasr-saab committed:
Merge branch 'main' of https://github.com/nasr-saab/Nasr_tutorial
